include/rdma/nccl_ofi_rdma_request.h

```diff
@@ -290,6 +291,68 @@ typedef struct nccl_net_ofi_rdma_req {
 	int (*free)(nccl_net_ofi_rdma_req_t *req,
 		    bool dec_inflight_reqs);
 
+
+	/* Get endpoint from request */
+	nccl_net_ofi_rdma_ep_t *rdma_req_get_ep();
+
+
+	/* Get device from request */
+	nccl_net_ofi_rdma_device_t *rdma_req_get_device();
+
+
+	/*
+	 * @brief	Return rx data struct of rx request
+	 */
+	rdma_req_rx_buff_data_t *get_rx_buff_data();
+	
+
+	/*
+	 * @brief	Return write inline struct of write request
+	 */
+	rdma_req_rma_op_data_t *req_get_rma_op_data(nccl_net_ofi_rdma_req_type_t req_type);
+
+
+	/*
+ 	 * @brief	Return send data struct of send request
+ 	 */
+	rdma_req_send_data_t *get_send_data();
+	
+	
+	/*
+ 	 * @brief	Return recv data struct of recv request
+ 	 */
+	rdma_req_recv_data_t *get_recv_data();
```

Comment:
oi, we have an ugly choice here.  These all really should only be in the send/recv/whatever requests, and not all in the base rdma request.  But we don't have that split today.